clusterctl/clusterdeployer/clusterdeployer_test.go

```diff
@@ -109,24 +125,66 @@ func (c *testClusterClient) GetClusterObjects() ([]*clusterv1.Cluster, error) {
 	return c.clusters, c.GetClusterObjectsErr
 }
 
+func (c *testClusterClient) GetMachineDeploymentObjects() ([]*clusterv1.MachineDeployment, error) {
```

Comment:
Should this return a copy? What happens if, after the callers holds on to it for a while, it is mutated? Ditto for `GetMachineSetObjects()`.